reactioncommerce/reaction
# (fix) #4408 Can't save discount codes

Resolves #4408   
Impact: **major**  
Type: **bugfix**

## Issue
Discount codes can't be saved because `shopId` is a required field but is not present in the document being submitted. I checked the method `"discounts/addCode` where `shopId` is being added to the document but it seems that `SimpleSchema` validation comes first before the document hits the said method causing the error.

## Solution
Include `shopId` in the form.

## Breaking changes
None.

## Testing
1. Go to Payments.
2. Enable Codes and create a discount code.
3. Save the discount code. There shouldn't be a validation error.
4. Edit existing code. The changes made should persist.
